components/brave_news/browser/resources/Feed.tsx

```diff
@@ -145,18 +151,12 @@ export default function Component({ feed, hasSubscriptions }: Props) {
     })
   }, [cardCount, feed?.items])
 
-  // An empty feed may still have ads in it, so we need to filter them out to
-  // determine if there are no articles.
-  const noArticles = React.useMemo(() => !feed?.items.filter(i => !i.advert).length, [feed])
-
   return <FeedContainer className={NEWS_FEED_CLASS}>
     {feed
-      ? noArticles
-        ? hasSubscriptions ? <NoArticles /> : <NoFeeds />
-        : <>
-          {cards}
-          <CaughtUp />
-        </>
+      ? errors[feed.error!] ?? <>
```

Comment:
I'm curious if this is a hacky way to do the same with
```
(feed.error && errors[feed.error]) ??
```
, as `errors[null]` would be undefined anyway

Reply:
Exactly - errors has an exact list of keys it allows, but I wanted to treat it as a map which may or may not have the key. Just felt it read a bit nicer, but happy to change it :smile: